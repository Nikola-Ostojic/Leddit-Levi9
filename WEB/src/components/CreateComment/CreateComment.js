import React, { useState } from "react";
import { Button, TextField } from "@material-ui/core";
import { toast } from "react-toastify";

import { apiCall } from "utils/ApiCall";

const CreateComment = ({ threadId, onThreadCreated }) => {
  const [content, setContent] = useState("");

  const createComment = (e) => {
    e.preventDefault();
    apiCall({
      url: "/comments",
      method: "POST",
      data: { content, threadId: parseInt(threadId) },
      requiresAuth: true
    })
      .then((response) => {
        toast.success(`Comment: ${content} created!`);
        // Cleaning up the state after a successful creation
        setContent("");
        onThreadCreated();
      })
      .catch((e) => {
        toast.error("Failed to create comment");
      });
  };

  return (
    <form onSubmit={createComment}>
      <TextField
        fullWidth
        margin="normal"
        label="Comment content"
        type="text"
        required
        onChange={({ target: { value } }) => setContent(value)}
        value={content}
      />
      <Button variant="contained" color="primary" type="submit">
        Create comment
      </Button>
    </form>
  );
};

export default CreateComment;
