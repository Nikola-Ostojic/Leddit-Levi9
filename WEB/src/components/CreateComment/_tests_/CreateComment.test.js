import React from "react";
import { shallow } from "enzyme";

import CreateComment from "../CreateComment";

describe("CreateComment component tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<CreateComment />);

    expect(wrapper).toMatchSnapshot();
  });
});
