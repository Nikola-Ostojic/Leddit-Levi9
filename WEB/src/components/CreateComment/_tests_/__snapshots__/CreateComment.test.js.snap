// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateComment component tests Should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    label="Comment content"
    margin="normal"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <WithStyles(ForwardRef(Button))
    color="primary"
    type="submit"
    variant="contained"
  >
    Create comment
  </WithStyles(ForwardRef(Button))>
</form>
`;
